### PR TITLE
Pull Request Summary for Global stock screener

### PR DESCRIPTION
**Pull request by the-faizmohammad**
This pull request outlines the requirements and guidelines for a web application development project. Here is a summary of the key points:

**Project Requirements:**
- The project requires the selection of an API that provides numeric values based on a parameter. The API can be open or require an API key.
- Examples of potential APIs include a stock data API or an air quality data API.
- Design guidelines should be followed, including selecting a main color, specifying typography (font face, size, and weight), and defining the layout and spacing between elements. The design is based on an original idea by Nelson Sakwa on Behance, and proper credit must be given in the README due to the Creative Commons license.
- The web application should have interactions, including a home page displaying categories that can be filtered by a parameter. Numeric values from the API should be displayed for each category.
- Clicking on a category item should navigate the user to a details page.
- On the details page, detailed data from the API for the selected category should be displayed.
- A "Back" button should allow users to return to the home page.
- Testing requirements include creating unit tests for Redux code and integration tests using the React Testing Library. Mocking API access and Redux Store connections may be necessary for testing.
- The project should be a single-page application (SPA) built with React and Redux.
- Data retrieved from the API should be stored in the Redux store.
- Filtering of API data should be done using a Filter stateless component.
- Each page within the SPA should have a unique route.
- The project should be deployed and accessible online, with a link to the online version included in the README.
- Project documentation should include a 5-minute video presentation that describes the project, demonstrates its features, and highlights noteworthy code or achievements.

In summary, the pull request outlines the technical and design requirements for building a web application, along with testing and documentation guidelines to ensure the successful development and presentation of the project.

### CODE REVIEW

File: src/components/Navigation.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Link } from 'react-router-dom';
+import './styles/Navigation.css';
+
+function Navigation() {
+  return (
+    <ul className="nav-bar">
```
Comment (shahadat3669) [inline src/components/Navigation.js]: - [ ] Your navbar component is a great addition to your app; however, it currently doesn't align with the provided design. To enhance your app's aesthetics and user experience, please ensure that you use the specified colors, pay attention to font styling, adjust spacing and layout to match the design, and include icons and dynamic text where required. By incorporating these design elements, you'll bring your navbar closer to the intended design, improving the overall look and feel of your app. Keep up the good work! 👍🎨

## Design (Navbar)

image

_**please follow the design for both the home page and the details page.**_

File: src/index.js [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
+  reducer: rootReducer,
+});
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
   <React.StrictMode>
-    <App />
-  </React.StrictMode>
+    <Provider store={store}>
+      <App />
```
Comment (shahadat3669) [inline src/index.js:17]: - [ ] Excellent work on implementing all the functionalities in your app; however, there's room for improvement in terms of design consistency. The color choices, background colors, and font families throughout your app do not align with the specified design. It's crucial to adhere to the design guidelines for both visual appeal and best practices. I recommend dedicating some time to systematically address these design elements one by one to bring your app in line with the intended look and feel. This effort will enhance the overall user experience and make your app more polished. Keep up the great work! 👍🎨



|   | Your | Design |
|  --- | --- | --- |
| Home  | image | image|
| Details  | image | image |

File: src/components/Company.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  }, [dispatch]);
+
+  return (
+    <div className="navigation">
+      <ul className="nav-bar">
+        <li className="subject">Stock Screener</li>
+        <li>
+          <Search />
+        </li>
+      </ul>
```
Comment (shahadat3669) [inline src/components/Company.js]: - [ ] Great effort on implementing the app's functionalities! However, to enhance the user experience, it's essential to align the visual aspects with the design guidelines. Currently, the background colors, font styles, and font sizes needed does not match the specified design. Please consider adjusting your code to match the design. Additionally, the top bar's font size should be adjusted to align more closely with the design. 

- [ ] To achieve a design-consistent look, consider adding the visual separation between the navbar and the topbar, as indicated in the design. These refinements will bring your app closer to the intended design aesthetics and create a more polished user interface. Keep up the good work! 👍🎨

| Your | Design |
| --- | --- |
|  image| image|

File: src/components/Company.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                <ul className="company-list" key={screener.symbol}>
+                  <li className="companySymbol">{screener.symbol}</li>
+                  <li style={{ marginBottom: '50px' }}>{screener.country}</li>
+                  <li className="company-name">{screener.companyName}</li>
+                  <li>{screener.sector}</li>
+                </ul>
+              </Link>
+            </div>
+          ))}
+      </div>
```
Comment (shahadat3669) [inline src/components/Company.js]: 👍 Great progress on your project! However, there are a couple of design-related improvements that can be made:

- [ ] Currently, there is some spacing between the cards, but according to the design, there should be no gap between the cards. Please consider removing this space to align with the design.

- [ ] Additionally, the spacing on the right and left sides appears to be larger than what's shown in the design. Kindly adjust your code to match the design specifications more closely.

| Your | Design |
| --- | --- |
|  image| image|


By making these adjustments, you'll bring your project even closer to the intended design. Keep up the good work! 👍🎨

File: src/components/Company.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+              <Link
+                to="/Screeners"
+                onClick={() => dispatch(selectCompany(screener))}
+              >
+                <ul className="company-list" key={screener.symbol}>
+                  <li className="companySymbol">{screener.symbol}</li>
+                  <li style={{ marginBottom: '50px' }}>{screener.country}</li>
+                  <li className="company-name">{screener.companyName}</li>
+                  <li>{screener.sector}</li>
+                </ul>
```
Comment (shahadat3669) [inline src/components/Company.js]: Great effort on implementing the company data display! However, there are a few enhancements to consider:

1. 👉 **Company Image**: As per the design, you can showcase the company image. You can achieve this by dynamically generating the image URL using the company symbol provided by the API. For example, for NVIDIA Corporation, the company image URL would be `https://financialmodelingprep.com/image-stock/NVDA.png`. This will add a visual element to your display where symbol is `NVDA`.

2. 📐 **Text Alignment**: In the design, the text is right-aligned, but currently, your text is center-aligned. Please consider aligning the text to the right to match the design specifications.

3. 🔢 **Numeric Values**: According to the project requirements, in addition to the text values, you should also display some numeric values per category that come from the API. Currently, you are displaying only text values. To enhance your project, kindly integrate and display the required numeric data as well.

4. align all elements as in the design. and there should be an icon at the top right corner as per design. Please consider adding that icon.

| Your | Design |
| --- | --- |
|  image| image |

Keep up the good work, and these enhancements will bring your project closer to the intended design and functionality! 👍🌟

File: src/routes/Screeners.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+            <span className="value">{stockScreeners.exchange}</span>
+          </li>
+          <li>
+            <span className="label">Beta:</span>
+            <span className="value">{stockScreeners.beta}</span>
+          </li>
+        </ul>
+
+      </div>
+    </div>
```
Comment (shahadat3669) [inline src/routes/Screeners.js]: Good job on your screen implementation! However, to align it better with the design, please consider the following improvements:

1. 🖼️ **Layout and Spacing**: The layout and spacing in your implementation do not fully match the design. Review the design carefully, paying attention to the placement and spacing of elements. Make adjustments to ensure your screen looks as close to the design as possible.

2. 📝 **Font Styling**: Check the font size and styling in your implementation. Ensure that they match the design specifications. Consistency in font styling is crucial for a polished appearance.

By making these adjustments, your screen will better align with the design, creating a more cohesive and visually appealing user interface. Keep up the good work! 👍🎨

File: src/routes/Screeners.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    fetchData();
+  }, [dispatch]);
+
+  return (
+    <div className="screenersContain">
+      {isLoading && <div>Loading...</div>}
+      <Link className="back-icon" to="/">
+        <HiChevronLeft />
+      </Link>
+      <div key={stockScreeners.symbol} className="screenersList">
```
Comment (shahadat3669) [inline src/routes/Screeners.js]: Nice work on the navbar! To further improve it and align with the design, please consider the following:

1. 📷 **Company Image**: As per the design, there should be a visual section displaying the company image. You can dynamically generate the image URL using the company's symbol, like this: `https://financialmodelingprep.com/image-stock/[symbol].png`. Replace `[symbol]` with the company's symbol provided by the API. Ensure that the image is displayed correctly.

2. 🖊️ **Layout and text Alignment**: In the design, the text is right-aligned, but in your implementation, it appears to be center-aligned. Adjust the text alignment and overall layout to match the design specifications.
3. Additionally, there should be a topbar before the details which is also missing in your code consider adding it to match the design.

| Your | Design |
| --- | --- |
|  image | image |

By making these changes, your navbar will better adhere to the design and provide a more visually consistent user experience. Great job, and keep up the good work! 👍🎨

File: src/routes/Screeners.js [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+          </li>
+          <li>
+            <span className="label">Exchange:</span>
+            <span className="value">{stockScreeners.exchange}</span>
+          </li>
+          <li>
+            <span className="label">Beta:</span>
+            <span className="value">{stockScreeners.beta}</span>
+          </li>
+        </ul>
```
Comment (shahadat3669) [inline src/routes/Screeners.js]: - [ ] Well done on implementing the UI list! To make it align more closely with the design, please consider these suggestions:

The font size in your UI list appears to be different from the design. Try adjusting the font size to match the design specifications for a consistent look.The background color in the list might need some adjustments to match the design's color scheme more accurately. The text alignment in your UI list may need modification to align it correctly with the design. Double-check the design for any specific alignment instructions. There should be an icon for each list item.
image





By incorporating these adjustments, your UI list will align better with the design, creating a more harmonious and visually appealing user interface. Keep up the good work! 👍🎨

File: README.md [hunk 9]
```diff
@@ -1,70 +1,160 @@
...
+
+Can I contribute to this project?\*\*
+
+- Yes, just create new feature branch.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 📝 License <a name="license"></a>
+
+This project is [LICENSED](./LICENSE) licensed.
```
Comment (shahadat3669) [inline README.md:158]: - [ ] Great job including a custom license file! However, it seems that the link to the license is currently broken. To ensure that users can access and review the license for this project, kindly update the link to the correct location. This will provide transparency and clarity regarding the licensing terms, which is essential for open-source projects. Keep up the good work! 👍📜

File: README.md [hunk 10]
```diff
@@ -1,70 +1,160 @@
...
+### Live Demo <a name="live-demo">
+https://global-stock-screener.onrender.com
 
-This section has moved here: [https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify](https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify)
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🚀 Loom Video Walkthrough <a name="video-walkthrough"></a>
+
+- [Loom Video Walkthrough](https://www.loom.com/share/9d3ff1f01ff942e481c790bc9cb9fb7f?sid=32f7e238-473e-462d-9014-cde462f62687)
```
Comment (shahadat3669) [inline README.md]: - [ ] Great job on your project presentation video! It's concise and effectively communicates the project's topic and purpose in professional. However, to make it even more informative and engaging, consider including a demonstration of the project where you showcase all its features. This will provide viewers with a clear understanding of how the project works and what it can do. Additionally, adding a section where you highlight a specific piece of code that you find interesting or are proud of creating can add a personal touch and give viewers insights into your coding skills and thought process. Keep up the excellent work! 👍📹

File: README.md [hunk 11]
```diff
@@ -1,70 +1,160 @@
...
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+I would like to thank Original design idea by Nelson Sakwa on Behance.
+
+- Microverse Review Team
+- My Coding Partners
+- Microverse Community in Slack
```
Comment (shahadat3669) [inline README.md:140]: - [ ] In your project, please make sure to include in  "Acknowledgements" section add credits [Nelson Sakwa](https://www.behance.net/sakwadesignstudio), the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as per the requirements of the [Creative Commons License](https://creativecommons.org/licenses/). This will ensure that you are in compliance with the licensing terms and show respect for the original creator's work. Great attention to detail! 👍📜

File: src/App.js
Comment (shahadat3669) [inline src/App.js:1]: - [ ] 👍 Great work on adding unit tests for your Redux slice! However, to fully meet the project requirements, kindly consider creating integration tests for your application using the React Testing Library. This will help ensure that your components are functioning correctly together and provide comprehensive test coverage for your application. Keep up the good work! 🧪🚀

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @the-faizmohammad , 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ No linter Errors
✅ Add unit test cases for the redux slice
✅ Used redux for state management.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by skyv26 — COMMENTED** [review]
## Hi @the-faizmohammad, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 🟨 



## Reason 🗒️ 
You submitted a project with the `wrong` gitflow. According to the instructions, you should create a pull request (PR) from the `feature` branch to the `development` branch, not from `development` to `main`. Since your project was `unapproved` last time, you need to make a fresh start. You can create a new branch that contains all the changes as `features` and then create a PR from that `feature` branch to the `development` branch.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

#### Cheers, and Happy coding!👏👏👏

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

--------

_Invalid Code Review Request **does not count** into the code reviews limit._
